File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums)
    {
        unordered_set<int> st;
        for (int i = 0; i < nums.size(); i++)
        {
            if (st.count(nums[i]) == 0)
            {
                st.insert(nums[i]);
            }
            else
            {
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        leftPointer = 0
        k = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[leftPointer] = nums[i]
                leftPointer += 1
                k += 1
        return k